app/components/components.md

```diff
@@ -0,0 +1,141 @@
+**ALL PAGES**
+
+* NAV component:
+- site name
+- cart icon
+- login button || username
+
+* FOOTER component?
+
+**PRODUCTS PAGE**
+
+* FILTER component:
+- Categories
+- Price ranges
+- keyword
+
+* ALL PRODUCTS component:
+- pagination
+- sort by
+- image
+- title
+- price
+- '+' (add to cart)
+- more info
+
+**INDIVIDUAL PRODUCT PAGE**
+
+* PRODUCT component:
+- image
+- title
+- price
+- description
+- add to cart (disabled if product is sold out)
+
+**CART PAGE**
+
+* CART component:
+- cart items
+- total cost
+- checkout
+
+* CARTITEM component:
+- image
+- title
+- price
+- quantity (editable)
+
+**LOGIN PAGE**
+
+* LOGIN FORM component:
+- email
+- password
+- submit
+
+* LOGIN ERROR component (optional)
+
+**ACCOUNT INFO PAGE**
+
+* ACCOUNT DETAILS component (form with filled in details that can be edited?):
+- first name
+- last name
+- email
+- password
+- profile pic?
+- preferences (this could be fun - the user could choose a default color scheme, display view, etc.)
+- order history
+- logout
+
+**ORDER HISTORY PAGE**
+
+* PAST ORDERS component:
+- past orders, with total price & status
+
+**PAST ORDER PAGE**
+
+* PAST ORDER component:
+- items
+- total cost
+- status
+- option to update (remove, update quantity) if order status is "pending"
+
+-----------
+
+ADMIN VIEW:
+
+**INDIVIDUAL PRODUCT PAGE**
+
+* PRODUCT component (editable):
+- name
+- description
+- inventory
+- price
+
+**ALL USERS PAGE**
+
+* ALL USERS component:
+- filter by type (guest, logged in, admin)
+- users (first name, last name, email, isRegistered, createdAt)
+
+**UPDATE USER PAGE**
+
+* USER component (form with filled in details that can be edited?):
+- first name
+- last name
+- email
+- password
+- profile pic?
+- preferences (this could be fun - the user could choose a default color scheme, display view, etc.)
+- delete user
+
+**ALL CARTS PAGE**
+
+* ALL CARTS component:
+- id
+- user id
+- user name
+- user email
+
+**INDIVIDUAL CART PAGE**
+
+* CART component:
+- cart items
+- total cost
+
+**ALL ORDERS PAGE**
+
+* ALL ORDERS component:
+- id
+- user id
+- total
+- status
+
+**INDIVIDUAL ORDER PAGE**
+
+* ORDER component (editable)
```

Comment:
Should admin be able to delete an order too?

Reply:
PS note that as per API routes, only the status on an order can be edited

Reply:
Noted. I think order items should be able to be updated. we can talk about that tonight

Reply:
Oh wait, order item already can be updated. let's discuss how a customer can delete an orderItem